src/daml/Daml/Trading/Cash/CashRedemption.daml

```diff
@@ -0,0 +1,82 @@
+module Daml.Trading.Cash.CashRedemption where
+
+import Daml.Finance.Interface.Holding.Base qualified as Holding (I)
+import Daml.Trading.Cash.CashTransfer qualified as CashTransfer
+import Daml.Finance.Interface.Types.Common (AccountKey, InstrumentKey(..))
+import Daml.Finance.Interface.Holding.Fungible (ArchiveFungible(ArchiveFungible))
+import Daml.Finance.Interface.Holding.Fungible qualified as Fungible (I, Split(..), SplitResult(..))
+import Daml.Finance.Interface.Holding.Util (getAmount)
+
+template SplitRequest
+  with
+    currentOwner: Party
+    amount: Decimal
+  where
+    signatory currentOwner
+    nonconsuming choice Split : ([ContractId Fungible.I], Optional (ContractId Fungible.I))
+      with
+        holdingCid : ContractId Holding.I
+      controller currentOwner
+      do
+      -- Sanity checks
+        holding <- fetch holdingCid
+        assertMsg "Balance insufficient" (amount <= getAmount holding)
+        let
+          fungibleCid : ContractId Fungible.I = coerceContractId holdingCid
+        Fungible.SplitResult{splitCids,rest} <- exercise fungibleCid Fungible.Split with amounts = [amount]
+        pure (splitCids, rest)
+
+
+template WithdrawalRequest
+  with
+    operator : Party
+    customer : Party
+    holdingCid : ContractId Holding.I
+    amount : Decimal
+    instrument: InstrumentKey
+  where
+    signatory customer
+    observer operator
+
+    ensure amount > 0.0
+    choice Accept : Optional (ContractId Holding.I)
+      with
+        adminBurnAccount : AccountKey
```

Comment:
I don't think this is needed since we can recreate the key from `custodian`, `owner`, and `cashLabel` (imported from Daml.Trading.Account.Account) inside the choice

Reply:
changed from input adminBurnAccount to adminBurnAccount = AccountKey with custodian = custodian, owner = owner, id = Id CreateAccount.cashLabel. Done